AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -7,25 +7,126 @@
+        }
+        
+        for layer in topSubLayer {
+            layer.frame = self.videoContainerTop.bounds
+        }
+        
+        for layer in bottomSubLayer {
+            layer.frame = self.videoContainerBottom.bounds
+        }
+        
+        view.layoutIfNeeded()
     }
     
     @IBAction func loadVideo(_ sender: UIButton) {
+        let imagePickerController = UIImagePickerController()
+        imagePickerController.sourceType = .photoLibrary
+        // recall that, by default, this array only contains static pix...
+        imagePickerController.mediaTypes = [String(kUTTypeMovie)]
+        imagePickerController.delegate = self
+        self.present(imagePickerController, animated: true, completion: nil)
+    }
     
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : Any]) {
+        if let url = info[UIImagePickerControllerReferenceURL] as? URL {
+            self.movieURL = url
+            
+            let playerItem = AVPlayerItem(url: url)
+            let player = AVPlayer(playerItem: playerItem)
+            let playerLayer = AVPlayerLayer(player: player)
+            
+            // looking at http://stackoverflow.com/questions/29000251/swift-resize-an-avplayerlayer-to-match-the-bounds-of-its-parent-container
+            
+            movies.append(playerLayer)
+            
+            if self.videoContainerTop.layer.sublayers == nil && self.videoContainerBottom.layer.sublayers == nil {
+                self.videoContainerTop.layer.addSublayer(playerLayer)
+                playerLayer.frame = videoContainerTop.bounds
+                player.play()
+            } else if videoContainerBottom.layer.sublayers == nil {
+                self.videoContainerBottom.layer.addSublayer(playerLayer)
+                playerLayer.frame = videoContainerBottom.bounds
+                player.play()
+            } else {
+                let top = movies[0]
+                let bottom = movies[1]
+                
+                if top.player?.rate != 0 && bottom.player?.rate == 0 {
+                    bottom.removeFromSuperlayer()
+                    movies[1] = playerLayer
+                    self.videoContainerBottom.layer.addSublayer(playerLayer)
+                    playerLayer.frame = videoContainerBottom.bounds
+                    player.play()
+                } else if top.player?.rate == 0 && bottom.player?.rate != 0 {
+                    top.removeFromSuperlayer()
+                    movies[0] = playerLayer
+                    self.videoContainerTop.layer.addSublayer(playerLayer)
+                    playerLayer.frame = videoContainerTop.bounds
+                    player.play()
+                } else {
+                    dismiss(animated: true) {
+                    if let url = self.movieURL {
+                        let player = AVPlayer(url: url)
+                        let playerController = AVPlayerViewController()
```

Comment:
The assignment didn't specify using ```AVPlayerViewController``` but to put the ```AVPlayer``` in and ```AVPlayerLayer``` which you did do. Here, you set up the playerController but never present or embed it so this code doesn't effect (or affect) anything the user sees.